src/xls_handler.py

```diff
@@ -76,189 +97,230 @@ def find_timetable(self, group_name, timetable):
-                            sheet.cell_value(main_row, main_col + iter)
-                        ) != "":
+                        cell_value_iter = sheet.cell_value(
+                            main_row,
+                            main_col + iter
+                        )
+                        if str(cell_value_iter) != "":
                             break
-                        if str(
-                            merged_dict.get((main_row, main_col + iter), "")
-                        ) != "" and str(
-                            merged_dict.get((main_row, main_col + iter), "")
-                        ) == sheet.cell_value(main_row, main_col):
+
+                        merged_cell = merged_dict.get(
+                            (main_row, main_col + iter),
+                            ""
+                        )
+
+                        if merged_cell != "" and merged_cell == cell_value:
                             groups += ", "
-                            groups += str(
-                                sheet.cell_value(3, main_col + iter)
-                            ) if str(
-                                sheet.cell_value(2, main_col + iter)
-                            ) == "" else str(
-                                sheet.cell_value(2, main_col + iter)
-                            ) + ", " + str(
-                                sheet.cell_value(3, main_col + iter)
-                            )
+                            cell_value_2 = sheet.cell_value(2, main_col + iter)
+                            cell_value_3 = sheet.cell_value(3, main_col + iter)
+                            if str(cell_value_2) == "":
+                                groups += str(cell_value_3)
+                            else:
+                                groups += str(cell_value_2)
+                                groups += ", "
+                                groups += str(cell_value_3)
+
                         iter += 1
 
-                    if res[
-                        request
-                    ][
-                        "week: {}".format(week)
-                    ][
-                        "day: {}".format(day)
-                    ][
-                        "{} lesson: ".format(lesson)
-                    ] == "":
-                        res[
-                            request
-                        ][
-                            "week: {}".format(week)
-                        ][
-                            "day: {}".format(day)
-                        ][
-                            "{} lesson: ".format(lesson)
-                        ] = str(
-                            sheet.cell_value(main_row, main_col)
-                        ) + " *" + groups + "*"
+                    request_dict = res[request]
+                    week_dict = request_dict[f"week: {week}"]
+                    day_dict = week_dict[f"day: {day}"]
+                    if day_dict[f"{lesson} lesson: "] == "":
```

Comment:
It'll work the same with `if not day_dict[f"{lesson} lesson: "]:`
And maybe it would be better to move `f"{lesson} lesson: "` to separate variable